# Data Preparation

This Jupyter Notebook cleans & prepares the raw Steam Games dataset for further analysis.

In [28]:
# Import libraries
import numpy as np
import pandas as pd
import re

from dateutil.parser import parse
from sklearn.preprocessing import MultiLabelBinarizer

In [29]:
games = pd.read_csv('datasets/raw_steam_games.csv')
games

url types  \
0        https://store.steampowered.com/app/379720/DOOM/   app   
1      https://store.steampowered.com/app/578080/PLAY...   app   
2      https://store.steampowered.com/app/637090/BATT...   app   
3        https://store.steampowered.com/app/221100/DayZ/   app   
4      https://store.steampowered.com/app/8500/EVE_On...   app   
...                                                  ...   ...   
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
0                                                   DOOM   
1                          PLAYERUNKNOWN'S BATTLEGROUNDS   
2                                             BATTLETECH   
3                                                   DayZ   
4                                             EVE Online   
...                                                  ...   
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                            desc_snippet  \
0      Now includes all three premium DLC packs (Unto...   
1      PLAYERUNKNOWN'S BATTLEGROUNDS is a battle roya...   
2      Take command of your own mercenary outfit of '...   
3      The post-soviet country of Chernarus is struck...   
4      EVE Online is a community-driven spaceship MMO...   
...                                                  ...   
40828                                                NaN   
40829                                                NaN   
40830                                                NaN   
40831                                                NaN   
40832  An RPG about a tribe exploring an ancient worl...   

                                          recent_reviews  \
0      Very Positive,(554),- 89% of the 554 user revi...   
1      Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2      Mixed,(166),- 54% of the 166 user reviews in t...   
3      Mixed,(932),- 57% of the 932 user reviews in t...   
4      Mixed,(287),- 54% of the 287 user reviews in t...   
...                                                  ...   
40828                                                NaN   
40829                                                NaN   
40830                                                NaN   
40831                                                NaN   
40832                                                NaN   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40828                                                NaN  Feb 12, 2019   
40829                                                NaN   Feb 5, 2019   
40830                                                NaN  Jul 31, 2018   
40831                                                NaN  Jul 24, 2018   
40832                                                NaN   Aug 8, 2018   

                     developer                                publisher  \
0                  id Software    Bethesda Softworks,Bethesda Softworks   
1             PUBG Corporation        PUBG Corporation,PUB

## Remove malformed rows

In [30]:
games = games[games['types'] == 'app'] # remove non-game rows
games = games.dropna(subset = ['all_reviews', 'release_date', 'languages']) # remove games that do not have `all_review`/`release_date`, i.e. unreleased games
games = games[~games['all_reviews'].str.contains('Need more user reviews to generate a score')] # remove games that do not have enough user reviews yet
games

url types  \
0        https://store.steampowered.com/app/379720/DOOM/   app   
1      https://store.steampowered.com/app/578080/PLAY...   app   
2      https://store.steampowered.com/app/637090/BATT...   app   
3        https://store.steampowered.com/app/221100/DayZ/   app   
4      https://store.steampowered.com/app/8500/EVE_On...   app   
...                                                  ...   ...   
40557  https://store.steampowered.com/app/652810/Grab...   app   
40598  https://store.steampowered.com/app/848410/Deta...   app   
40611  https://store.steampowered.com/app/454330/A_Ro...   app   
40728  https://store.steampowered.com/app/763990/Chas...   app   
40811  https://store.steampowered.com/app/809080/Nanoui/   app   

                                name  \
0                               DOOM   
1      PLAYERUNKNOWN'S BATTLEGROUNDS   
2                         BATTLETECH   
3                               DayZ   
4                         EVE Online   
...                              ...   
40557                        Grabity   
40598       Detached: Non-VR Edition   
40611                  A Room Beyond   
40728              Chasing the Stars   
40811                         Nanoui   

                                            desc_snippet  \
0      Now includes all three premium DLC packs (Unto...   
1      PLAYERUNKNOWN'S BATTLEGROUNDS is a battle roya...   
2      Take command of your own mercenary outfit of '...   
3      The post-soviet country of Chernarus is struck...   
4      EVE Online is a community-driven spaceship MMO...   
...                                                  ...   
40557  Dance the tango of death in Grabity, a fast-pa...   
40598  Detached, a suspenseful interstellar duel that...   
40611  Solve a mysterious criminal case in this dark ...   
40728  Chasing the Stars is a steampunk-ish multichoi...   
40811  The book of Nanoui has been stolen, plunging t...   

                                          recent_reviews  \
0      Very Positive,(554),- 89% of the 554 user revi...   
1      Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2      Mixed,(166),- 54% of the 166 user reviews in t...   
3      Mixed,(932),- 57% of the 932 user reviews in t...   
4      Mixed,(287),- 54% of the 287 user reviews in t...   
...                                                  ...   
40557                                                NaN   
40598                                                NaN   
40611                                                NaN   
40728                                                NaN   
40811                                                NaN   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40557  Positive,(17),- 100% of the 17 user reviews fo...  May 30, 2018   
40598  Mostly Positive,(14),- 78% of the 14 user revi...  Jul 24, 2018   
40611  Positive,(11),- 90% of the 11 user reviews for...  Jun 13, 2017   
40728  Mostly Positive,(10),- 70% of the 10 user revi...  Jan 23, 2019   
40811  Mostly Negative,(14),- 28% of the 14 user revi...  Mar 21, 2018   

                 developer                                publisher  \
0              id Software    Bethesda Softworks,Bethesda Softworks   
1         PUBG Corporation        PUBG Corporation,PUBG Corporation   
2      Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3      Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                      CCP                                  CCP,CCP   
...                   

## Remove unused columns

In [31]:
games = games.drop(['url', 'types', 'desc_snippet', 'recent_reviews', 'game_description', 'game_details', 'minimum_requirements', 'recommended_requirements', 'achievements'], axis = 1) # All these fields are either not useful or too difficult to extract useful information from.
games


name  \
0                               DOOM   
1      PLAYERUNKNOWN'S BATTLEGROUNDS   
2                         BATTLETECH   
3                               DayZ   
4                         EVE Online   
...                              ...   
40557                        Grabity   
40598       Detached: Non-VR Edition   
40611                  A Room Beyond   
40728              Chasing the Stars   
40811                         Nanoui   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40557  Positive,(17),- 100% of the 17 user reviews fo...  May 30, 2018   
40598  Mostly Positive,(14),- 78% of the 14 user revi...  Jul 24, 2018   
40611  Positive,(11),- 90% of the 11 user reviews for...  Jun 13, 2017   
40728  Mostly Positive,(10),- 70% of the 10 user revi...  Jan 23, 2019   
40811  Mostly Negative,(14),- 28% of the 14 user revi...  Mar 21, 2018   

                 developer                                publisher  \
0              id Software    Bethesda Softworks,Bethesda Softworks   
1         PUBG Corporation        PUBG Corporation,PUBG Corporation   
2      Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3      Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                      CCP                                  CCP,CCP   
...                    ...                                      ...   
40557    Team Ninja Thumbs      Team Ninja Thumbs,Team Ninja Thumbs   
40598       Anshar Studios            Anshar Studios,Anshar Studios   
40611         René Bühling                René Bühling,René Bühling   
40728          Ertal Games                  Ertal Games,Ertal Games   
40811             Phoesion                        Phoesion,Phoesion   

                                            popular_tags  \
0      FPS,Gore,Action,Demons,Shooter,First-Person,Gr...   
1      Survival,Shooter,Multiplayer,Battle Royale,PvP...   
2      Mechs,Strategy,Turn-Based,Turn-Based Tactics,S...   
3      Survival,Zombies,Open World,Multiplayer,PvP,Ma...   
4      Space,Massively Multiplayer,Sci-fi,Sandbox,MMO...   
...                                                  ...   
40557  Indie,Action,4 Player Local,Local Multiplayer,...   
40598  Simulation,Indie,Space,Exploration,Atmospheric...   
40611                      Adventure,Indie,Point & Click   
40728  Nudity,Sexual Content,Indie,Steampunk,Story Ri...   
40811  Casual,RPG,Action,Adventure,Indie,Violent,Co-o...   

                                               languages  \
0      English,French,Italian,German,Spanish - Spain,...   
1      English,Korean,Simplified Chinese,French,Germa...   
2                          English,French,German,Russian   
3      English,French,Italian,German,Spanish - Spain,...   
4                          English,German,Russian,French   
...                                                  ...   
40557   English,German,French,Portuguese,Spanish - Spain   
40598    English,French,German,Simplified Chinese,Polish   
40611      English,German,French,Italian,Spanish - Spain   
40728                                            English   
40811                                            English   

                                                   genre  \
0                                                 Action   
1                 Action,Adventure,Massively Multiplayer   
2                              Action,Adventure,Strategy   
3                 Action,Adventure,Massively Multiplayer   
4      Action,Free to Play,Massively Multiplayer,RPG,...   
...    

## Clean prices

In [32]:
pattern = re.compile('demo|prologue|prototype|try|trial|试玩|Guest', re.IGNORECASE) # remove all trial versions

def map_original_price(x):
    if pd.isna(x):
        return 0.0
    elif re.match(r'(\$\d+(\.\d{2})?)', x):
        return float(x[1:])
    else:
        return 0.0

mask = games['original_price'].str.contains(pattern, na=False)
games = games.loc[~mask]

games = games[~games['original_price'].isin(['1.020', '650560'])]  # remove rows with malformed prices
games['original_price'] = games['original_price'].apply(map_original_price)


def map_discount_price(x):
    if pd.isna(x):
        return x
    elif re.match(r'(\$\d+(\.\d{2})?)', x):
        return float(x[1:])
    else:
        return 0.0

games['discount_price'] = games['discount_price'].apply(map_discount_price)
games['discount_price'] = games['discount_price'].fillna(games['original_price'])
games = games[(games['original_price'] >= games['discount_price'])]
games['discount_percentage'] = (games['original_price'] - games['discount_price']) / games['original_price'] * 100
games['discount_percentage'] = games['discount_percentage'].fillna(0)


## Clean release dates

In [34]:
from datetime import datetime

unix_epoch = datetime(1970, 1, 1)

games['release_date'] = pd.to_datetime(games['release_date'], format='mixed')
games.insert(games.columns.get_loc('release_date') + 1, 'release_date_epoch_days', games['release_date'].apply(lambda x: (x - unix_epoch).days))
games


ValueError: time data 'May 12, 2016' does not match format 'mixed' (match)

## Clean genres

In [35]:
ignored = {
    # Non-game genres
    'Accounting', 'Animation & Modeling', 'Audio Production', 'Game Development', 'Movie', 'Photo Editing', 'Software Training', 'Utilities', 'Video Production', 'Web Publishing',
    'Design & Illustration', 'Education', 'Valve',
    # Already represented in original_price column
    'Free to Play',
    # These categories are not really "genre"s
    'Early Access', 'Indie',
}

def one_hot_encode_genres(frame):
    frame['genre'] = frame['genre'].apply(lambda x: [] if pd.isna(x) else [genre for genre in x.split(',') if genre not in ignored])
    frame = frame[frame['genre'].apply(len) > 0]

    binarizer = MultiLabelBinarizer(sparse_output = True)
    genres = pd.DataFrame.sparse.from_spmatrix(binarizer.fit_transform(frame['genre']), index=frame.index, columns=binarizer.classes_)
    genres = genres.add_prefix(f'genre_')

    frame = pd.concat([frame, genres], axis=1)
    return frame

games = one_hot_encode_genres(games)
games


name  \
0                               DOOM   
1      PLAYERUNKNOWN'S BATTLEGROUNDS   
2                         BATTLETECH   
3                               DayZ   
4                         EVE Online   
...                              ...   
40481   雨上がりのハナビィ Ameagari no Hanaby   
40524               Lil Big Invasion   
40598       Detached: Non-VR Edition   
40611                  A Room Beyond   
40811                         Nanoui   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40481  Positive,(30),- 100% of the 30 user reviews fo...  Sep 22, 2018   
40524  Positive,(13),- 100% of the 13 user reviews fo...   Aug 4, 2016   
40598  Mostly Positive,(14),- 78% of the 14 user revi...  Jul 24, 2018   
40611  Positive,(11),- 90% of the 11 user reviews for...  Jun 13, 2017   
40811  Mostly Negative,(14),- 28% of the 14 user revi...  Mar 21, 2018   

                 developer                                publisher  \
0              id Software    Bethesda Softworks,Bethesda Softworks   
1         PUBG Corporation        PUBG Corporation,PUBG Corporation   
2      Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3      Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                      CCP                                  CCP,CCP   
...                    ...                                      ...   
40481    Enigmatic Network      Enigmatic Network,Enigmatic Network   
40524      Andreas Britten          Andreas Britten,Andreas Britten   
40598       Anshar Studios            Anshar Studios,Anshar Studios   
40611         René Bühling                René Bühling,René Bühling   
40811             Phoesion                        Phoesion,Phoesion   

                                            popular_tags  \
0      FPS,Gore,Action,Demons,Shooter,First-Person,Gr...   
1      Survival,Shooter,Multiplayer,Battle Royale,PvP...   
2      Mechs,Strategy,Turn-Based,Turn-Based Tactics,S...   
3      Survival,Zombies,Open World,Multiplayer,PvP,Ma...   
4      Space,Massively Multiplayer,Sci-fi,Sandbox,MMO...   
...                                                  ...   
40481  Indie,Adventure,Action,Casual,Sexual Content,Cute   
40524  Action,Indie,Adventure,Puzzle,Atmospheric,Cute...   
40598  Simulation,Indie,Space,Exploration,Atmospheric...   
40611                      Adventure,Indie,Point & Click   
40811  Casual,RPG,Action,Adventure,Indie,Violent,Co-o...   

                                               languages  \
0      English,French,Italian,German,Spanish - Spain,...   
1      English,Korean,Simplified Chinese,French,Germa...   
2                          English,French,German,Russian   
3      English,French,Italian,German,Spanish - Spain,...   
4                          English,German,Russian,French   
...                                                  ...   
40481                                   English,Japanese   
40524                                            English   
40598    English,French,German,Simplified Chinese,Polish   
40611      English,German,French,Italian,Spanish - Spain   
40811                                            English   

                                                genre  \
0                                            [Action]   
1          [Action, Adventure, Massively Multiplayer]   
2                       [Action, Adventure, Strategy]   
3          [Action, Adventure, Massively Multiplayer]   
4      [Action, Massively Multiplayer, RPG, Strategy]   
...                      

## Clean developers & publishers

In [40]:
games.dropna(subset=['developer', 'publisher'], how='all', inplace=True) # remove malformed rows

# Replace NaN values, this is often the case where the developer & publisher are the same
games['developer'] = games['developer'].fillna(games['publisher'])
games['publisher'] = games['publisher'].fillna(games['developer'])
games['publisher'] = games['publisher'].str.split(',').str.get(0)
games

name  all_reviews  recommend  release_date  \
0                               DOOM           92       True  May 12, 2016   
1      PLAYERUNKNOWN'S BATTLEGROUNDS           49      False  Dec 21, 2017   
2                         BATTLETECH           71       True  Apr 24, 2018   
3                               DayZ           61      False  Dec 13, 2018   
4                         EVE Online           74       True   May 6, 2003   
...                              ...          ...        ...           ...   
40481   雨上がりのハナビィ Ameagari no Hanaby          100       True  Sep 22, 2018   
40524               Lil Big Invasion          100       True   Aug 4, 2016   
40598       Detached: Non-VR Edition           78       True  Jul 24, 2018   
40611                  A Room Beyond           90       True  Jun 13, 2017   
40811                         Nanoui           28      False  Mar 21, 2018   

                 developer            publisher  \
0              id Software   Bethesda Softworks   
1         PUBG Corporation     PUBG Corporation   
2      Harebrained Schemes  Paradox Interactive   
3      Bohemia Interactive  Bohemia Interactive   
4                      CCP                  CCP   
...                    ...                  ...   
40481    Enigmatic Network    Enigmatic Network   
40524      Andreas Britten      Andreas Britten   
40598       Anshar Studios       Anshar Studios   
40611         René Bühling         René Bühling   
40811             Phoesion             Phoesion   

                                            popular_tags  \
0      [FPS, Gore, Action, Demons, Shooter, First-Per...   
1      [Survival, Shooter, Multiplayer, Battle Royale...   
2      [Mechs, Strategy, Turn-Based, Turn-Based Tacti...   
3      [Survival, Zombies, Open World, Multiplayer, P...   
4      [Space, Massively Multiplayer, Sci-fi, Sandbox...   
...                                                  ...   
40481  [Indie, Adventure, Action, Casual, Sexual Cont...   
40524  [Action, Indie, Adventure, Puzzle, Atmospheric...   
40598  [Simulation, Indie, Space, Exploration, Atmosp...   
40611                  [Adventure, Indie, Point & Click]   
40811  [Casual, RPG, Action, Adventure, Indie, Violen...   

                                               languages  \
0      [English, French, Italian, German, Spanish - S...   
1      [English, Korean, Simplified Chinese, French, ...   
2                     [English, French, German, Russian]   
3      [English, French, Italian, German, Spanish - S...   
4                     [English, German, Russian, French]   
...                                                  ...   
40481                                [English, Japanese]   
40524                                          [English]   
40598  [English, French, German, Simplified Chinese, ...   
40611  [English, German, French, Italian, Spanish - S...   
40811                                          [English]   

                                                genre  mature_content  ...  \
0                                            [Action]           False  ...   
1          [Action, Adventure, Massively Multiplayer]            True  ...   
2                       [Action, Adventure, Strategy]           False  ...   
3          [Action, Adventure, Massively Multiplayer]           False  ...   
4      [Action, Massively Multiplayer, RPG, Strategy]           False  ...   
...                                               ...             ...  ...   
40481                     [Action, Adventure, Casual]            True  ...   
40524                             [Action, Adventure]           False  ...   
40598                                    [Simulation]           False  ...   
40611                                     [Adventure]           False  ...   
40811                [Action, Adventure, Casual, RPG]           False  ...   

       genre_Action  genre_Adventure  genre_Casual  \
0                 1                0             

## Finding the bar to set recommended games base on all_reviews score

In [37]:


pattern = r'(\d+)%'


# extract percentage from Sentiment column using regular expressions
games['Percentage'] = games['all_reviews'].str.extract(pattern)


# convert percentage column to float
games['Percentage'] = games['Percentage'].astype(float)


# use str.contains to count number of rows with sentiment of Very Positive or Mostly Positive, and percentage above 70%
count = len(games[(games['all_reviews'].str.contains('Positive')) & (games['Percentage'] >= 70)])
print("Number of rows with sentiment of Positive, and percentage above 70%:", count)

count = len(games[(games['all_reviews'].str.contains('Positive')) & (games['Percentage'] < 70)])
print("Number of rows with sentiment of Positive, and percentage below 70%:", count)

count = len(games[((games['all_reviews'].str.contains('Mixed')) |(games['all_reviews'].str.contains('Negative')))& (games['Percentage'] >= 70)])
print("Number of rows with sentiment of Mixed and Negative, and percentage above 70%:", count)

count = len(games[((games['all_reviews'].str.contains('Mixed')) |(games['all_reviews'].str.contains('Negative')))& (games['Percentage'] < 70)])
print("Number of rows with sentiment of Mixed and Negative, and percentage below 70%:", count)

Number of rows with sentiment of Positive, and percentage above 70%: 7550
Number of rows with sentiment of Positive, and percentage below 70%: 0
Number of rows with sentiment of Mixed and Negative, and percentage above 70%: 0
Number of rows with sentiment of Mixed and Negative, and percentage below 70%: 4070


We can see that all games are accounted for 7550+4070=11620 which the amount of rows of games from the cleaning done above.
Showing that the bar for games with positive connotatin are indeed set at 70% of positive reviews.

## Clean other data

In [38]:
def map_all_reviews(x):
    match = re.search(r"\d{1,3}%", x)
    if match:
        return int(match.group(0)[:-1])

games['all_reviews'] = games['all_reviews'].apply(map_all_reviews)
games.insert(games.columns.get_loc('all_reviews') + 1, 'recommend', games['all_reviews'].apply(lambda x: x >= 70))

games['languages'] = games['languages'].apply(lambda x: [] if pd.isna(x) else list(x.split(',')))
games['popular_tags'] = games['popular_tags'].apply(lambda x: [] if pd.isna(x) else list(x.split(',')))
games['mature_content'] = games['mature_content'].apply(lambda x: not pd.isna(x))
games['publisher'] = games['publisher'].str.split(',').str.get(0)
games

name  all_reviews  recommend  release_date  \
0                               DOOM           92       True  May 12, 2016   
1      PLAYERUNKNOWN'S BATTLEGROUNDS           49      False  Dec 21, 2017   
2                         BATTLETECH           71       True  Apr 24, 2018   
3                               DayZ           61      False  Dec 13, 2018   
4                         EVE Online           74       True   May 6, 2003   
...                              ...          ...        ...           ...   
40481   雨上がりのハナビィ Ameagari no Hanaby          100       True  Sep 22, 2018   
40524               Lil Big Invasion          100       True   Aug 4, 2016   
40598       Detached: Non-VR Edition           78       True  Jul 24, 2018   
40611                  A Room Beyond           90       True  Jun 13, 2017   
40811                         Nanoui           28      False  Mar 21, 2018   

                 developer            publisher  \
0              id Software   Bethesda Softworks   
1         PUBG Corporation     PUBG Corporation   
2      Harebrained Schemes  Paradox Interactive   
3      Bohemia Interactive  Bohemia Interactive   
4                      CCP                  CCP   
...                    ...                  ...   
40481    Enigmatic Network    Enigmatic Network   
40524      Andreas Britten      Andreas Britten   
40598       Anshar Studios       Anshar Studios   
40611         René Bühling         René Bühling   
40811             Phoesion             Phoesion   

                                            popular_tags  \
0      [FPS, Gore, Action, Demons, Shooter, First-Per...   
1      [Survival, Shooter, Multiplayer, Battle Royale...   
2      [Mechs, Strategy, Turn-Based, Turn-Based Tacti...   
3      [Survival, Zombies, Open World, Multiplayer, P...   
4      [Space, Massively Multiplayer, Sci-fi, Sandbox...   
...                                                  ...   
40481  [Indie, Adventure, Action, Casual, Sexual Cont...   
40524  [Action, Indie, Adventure, Puzzle, Atmospheric...   
40598  [Simulation, Indie, Space, Exploration, Atmosp...   
40611                  [Adventure, Indie, Point & Click]   
40811  [Casual, RPG, Action, Adventure, Indie, Violen...   

                                               languages  \
0      [English, French, Italian, German, Spanish - S...   
1      [English, Korean, Simplified Chinese, French, ...   
2                     [English, French, German, Russian]   
3      [English, French, Italian, German, Spanish - S...   
4                     [English, German, Russian, French]   
...                                                  ...   
40481                                [English, Japanese]   
40524                                          [English]   
40598  [English, French, German, Simplified Chinese, ...   
40611  [English, German, French, Italian, Spanish - S...   
40811                                          [English]   

                                                genre  mature_content  ...  \
0                                            [Action]           False  ...   
1          [Action, Adventure, Massively Multiplayer]            True  ...   
2                       [Action, Adventure, Strategy]           False  ...   
3          [Action, Adventure, Massively Multiplayer]           False  ...   
4      [Action, Massively Multiplayer, RPG, Strategy]           False  ...   
...                                               ...             ...  ...   
40481                     [Action, Adventure, Casual]            True  ...   
40524                             [Action, Adventure]           False  ...   
40598                                    [Simulation]           False  ...   
40611                                     [Adventure]           False  ...   
40811                [Action, Adventure, Casual, RPG]           False  ...   

       genre_Action  genre_Adventure  genre_Casual  \
0                 1                0             

In [39]:
# We store the results as Pickle instead of CSV since it allows us to preserve data structures, i.e. sparse lists.
games.to_pickle('datasets/cleaned_steam_games.pkl')

# To read the file, `df = pd.read_pickle('datasets/cleaned_steam_games.pkl')`